src/utils/task_stats_alloc/src/lib.rs

```diff
@@ -57,33 +71,55 @@ impl TaskLocalBytesAllocated {
     /// The caller must ensure that `self` is valid.
     #[inline(always)]
     unsafe fn add_unchecked(&self, val: usize) {
-        self.0.unwrap_unchecked().fetch_add(val, Ordering::Relaxed);
+        let bytes = self.0.unwrap_unchecked();
+        let bytes_ref = unsafe { bytes.as_ref() };
+        bytes_ref.fetch_add(val, Ordering::Relaxed);
     }
 
     /// Subtracts from the counter value, and `drop` the counter while the count reaches zero.
     #[inline(always)]
-    fn sub(&self, val: usize) {
+    pub(crate) fn sub(&self, val: usize) {
```

Comment:
An idea of testing whether the box is dropped:
Let this function return a bool indicating whether it is the last one.
```suggestion
    pub(crate) fn sub(&self, val: usize) -> bool {
```
Collect these values via thread join handles, and then assert that there is exact one `true` of them.

When used in real case, because the return value is unused and the function is always inline, it should be optimized out and cause no overhead.

Reply:
Good. Updated